ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -236,15 +238,28 @@ type FSharpOverridingMembersBuilder() =
                 let diff = m.Indent - indentationTarget
                 if diff > 0  && isFirstMeaningfulNodeOnLine m then shiftWithWhitespaceBefore -diff m)
 
-        // members etc. might need indenting or deindenting after the representation start was moved to it's own line
         let reprNodes =
-            typeRepr.Children()
-            |> Seq.filter
-                   (fun c -> c :? ICommentNode || c :? ITypeBodyMemberDeclaration || c :? IRecordFieldDeclarationList)
+            let children = typeRepr.Children() |> Seq.toList
+            let startIdx = children |> List.tryFindIndex (fun c -> c.NodeType = FSharpTokenType.CLASS
+                                                                    || c.NodeType = FSharpTokenType.STRUCT
+                                                                    || c.NodeType = FSharpTokenType.LBRACE)
```

Comment:
I think it could be easier to copy the approach you've used for normalizing the representations, where you get `beginToken` and `endToken`. It should be possible to reuse that.

```fsharp
children
|> Seq.skipWhile ((!=) beginToken)
|> Seq.tail
|> Seq.takeWhile ((!=) endToken) // assuming there's a `null` check for `endToken`
```

Reply:
Another approach could be this:
```fsharp
TreeRange(beginToken.NextSibling, endToken.PrevSibling)
```